templates/CRM/Core/Page/PaymentPage.tpl

```diff
@@ -4,6 +4,8 @@
   {/foreach}
   {foreach from=$display_fields key=display_field item=field_spec}
     {assign var='core_field_name' value=$field_spec.core_field_name}
+    {assign var='core_field_name_m' value=$field_spec.core_field_name_m}
```

Comment:
Hmm OK so we are kinda hacked here already for Cybersource (sorry about that). It seems to me that if we had a way other than        {elseif $field_spec.htmlType == 'date'} of identifying the wierdo core field we would be doing better. 
What about 
- adding metadata for the expiry_date_month & year to the payment_fields_metadata function 
-  giving them a different html type (Select) & just bypassing all that date messiness. 
- would required adding their options in in the metadata

The whole civicrm form date thing  It does seem to add a form rule to ensure a future date but I suspect that might not be functioning for you anyway